src/linker/Linker/LinkerAttributesInformation.cs

```diff
@@ -80,5 +108,80 @@ static Attribute ProcessRequiresUnreferencedCodeAttribute (LinkContext context,
 
 			return null;
 		}
+
+		static DynamicDependency ProcessDynamicDependencyAttribute (LinkContext context, IMemberDefinition member, CustomAttribute customAttribute)
+		{
+			if (!ShouldProcessDependencyAttribute (context, customAttribute))
+				return null;
+
+			var dynamicDependency = GetDynamicDependency (context, customAttribute);
+			if (dynamicDependency == null) {
+				context.LogMessage (MessageContainer.CreateWarningMessage ($"Invalid DynamicDependencyAttribute on '{member}'", 2030));
+				return null;
+			}
+
+			dynamicDependency.OriginalAttribute = customAttribute;
+			return dynamicDependency;
+		}
+
+		static DynamicDependency GetDynamicDependency (LinkContext context, CustomAttribute ca)
+		{
+			var args = ca.ConstructorArguments;
+			if (args.Count > 3)
+				return null;
+
+			// First argument is string or DynamicallyAccessedMemberTypes
+			string memberSignature = args[0].Value as string;
+			if (args.Count == 1)
+				return memberSignature == null ? null : new DynamicDependency (memberSignature);
+			DynamicallyAccessedMemberTypes? memberTypes = null;
+			if (memberSignature == null) {
+				var argType = args[0].Type;
+				if (!(argType.Namespace == "System.Diagnostics.CodeAnalysis" && argType.Name == "DynamicallyAccessedMemberTypes"))
+					return null;
+				try {
+					memberTypes = (DynamicallyAccessedMemberTypes) args[0].Value;
+				} catch (InvalidCastException) { }
+				if (memberTypes == null)
+					return null;
+			}
+
+			// Second argument is Type for ctors with two args, string for ctors with three args
+			if (args.Count == 2) {
+				if (!(args[1].Value is TypeReference type))
+					return null;
+				return memberSignature == null ? new DynamicDependency (memberTypes.Value, type) : new DynamicDependency (memberSignature, type);
+			}
+			Debug.Assert (args.Count == 3);
+			if (!(args[1].Value is string typeName))
+				return null;
+
+			// Third argument is assembly name
+			if (!(args[2].Value is string assemblyName))
+				return null;
+
+			return memberSignature == null ? new DynamicDependency (memberTypes.Value, typeName, assemblyName) : new DynamicDependency (memberSignature, typeName, assemblyName);
+		}
+
+		public static bool ShouldProcessDependencyAttribute (LinkContext context, CustomAttribute ca)
```

Comment:
Why not have this as a static member on the DynamicDependency class - I don't feel it belongs here really.

Reply:
Moved